compiler/damlc/tests/src/DA/Test/ShakeIdeClient.hs

```diff
@@ -299,29 +299,31 @@ dlintSmokeTests mbScenarioService = Tasty.testGroup "Dlint smoke tests"
             setFilesOfInterest [foo]
             expectNoErrors
             expectDiagnostic DsInfo (foo, 2, 0) "Warning: Use fewer imports"
-    ,  testCase' "Reduce duplication" $ do
-            foo <- makeFile "Foo.daml" $ T.unlines
-                [ "daml 1.2"
-                , "module Foo where"
-                , "import DA.List"
-                , "testSort5 = scenario do"
-                , "    let l = [ (2, const \"D\"), (1, const \"A\"), (1, const \"B\"), (3, const \"E\"), (1, const \"C\") ]"
-                , "        m = sortOn fst l"
-                , "        n = map fst m"
-                , "    assert $ n == [1, 1, 1, 2, 3]"
-                , "    let o = map (flip snd ()) m"
-                , "    assert $ o == [\"A\", \"B\", \"C\", \"D\", \"E\"]"
-                , "testSort4 = scenario do"
-                , "    let l = [ (2, const \"D\"), (1, const \"A\"), (1, const \"B\"), (3, const \"E\"), (1, const \"C\") ]"
-                , "        m = sortBy (\\x y -> compare (fst x) (fst y)) l"
-                , "        n = map fst m"
-                , "    assert $ n == [1, 1, 1, 2, 3]"
-                , "    let o = map (flip snd ()) m"
-                , "    assert $ o == [\"A\", \"B\", \"C\", \"D\", \"E\"]"
-                ]
-            setFilesOfInterest [foo]
-            expectNoErrors
-            expectDiagnostic DsInfo (foo, 7, 4) "Suggestion: Reduce duplication"
+    -- This hint is now disabled. See PR
+    -- https://github.com/digital-asset/daml/pull/6423 for details.
+    -- ,  testCase' "Reduce duplication" $ do
+    --         foo <- makeFile "Foo.daml" $ T.unlines
+    --             [ "daml 1.2"
+    --             , "module Foo where"
+    --             , "import DA.List"
+    --             , "testSort5 = scenario do"
+    --             , "    let l = [ (2, const \"D\"), (1, const \"A\"), (1, const \"B\"), (3, const \"E\"), (1, const \"C\") ]"
+    --             , "        m = sortOn fst l"
+    --             , "        n = map fst m"
+    --             , "    assert $ n == [1, 1, 1, 2, 3]"
+    --             , "    let o = map (flip snd ()) m"
+    --             , "    assert $ o == [\"A\", \"B\", \"C\", \"D\", \"E\"]"
+    --             , "testSort4 = scenario do"
+    --             , "    let l = [ (2, const \"D\"), (1, const \"A\"), (1, const \"B\"), (3, const \"E\"), (1, const \"C\") ]"
+    --             , "        m = sortBy (\\x y -> compare (fst x) (fst y)) l"
+    --             , "        n = map fst m"
+    --             , "    assert $ n == [1, 1, 1, 2, 3]"
+    --             , "    let o = map (flip snd ()) m"
+    --             , "    assert $ o == [\"A\", \"B\", \"C\", \"D\", \"E\"]"
+    --             ]
+    --         setFilesOfInterest [foo]
+    --         expectNoErrors
+    --         expectDiagnostic DsInfo (foo, 7, 4) "Suggestion: Reduce duplication"
```

Comment:
I suggest we completely delete this code. This hint has never done anything helpful for us and I doubt it will ever do.